addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -114,16 +114,27 @@ publicWidget.registry.popup = PopupWidget;
 // Prevent bootstrap to prevent scrolling and to add the strange body
 // padding-right they add if the popup does not use a backdrop (especially
 // important for default cookie bar).
+
+const _baseShowElement = $.fn.modal.Constructor.prototype._showElement;
+$.fn.modal.Constructor.prototype._showElement = function (relatedTarget){
+    _baseShowElement.apply(this, relatedTarget);
+    this._setScrollbar();
+}
+
 const _baseSetScrollbar = $.fn.modal.Constructor.prototype._setScrollbar;
 $.fn.modal.Constructor.prototype._setScrollbar = function () {
-    if (this._element.classList.contains('s_popup_no_backdrop')) {
+    const modalContent = this._element.getElementsByClassName('modal-content')[0];
+    if (this._element.classList.contains('s_popup_no_backdrop') && modalContent.offsetHeight < window.innerHeight) {
+        this._element.classList.remove('s_popup_overflow_page');
         return;
     }
+    this._element.classList.add('s_popup_overflow_page');
```

Comment:
Note that this class is also added during edit mode, I think the edit mode code should maybe remove it before save (cleanForSave?). Deep test this a bit more: even though it seems to fix the issue, I am still able to reproduce the issue if I try multiple times...

Reply:
Hi Quentin, I'm fixing the issues listed above, but I didn't understand this change. Why the edit mode should remove the class on save? Also regarding the issue, I can't find a way to reproduce it again outside edit mode. In edit mode if I show the cookie panel and then add blocks until they overflow the page, I still can't scroll. This happens cause _showElement isn't called each time a block is added or removed. Hiding and showing the cookie panel again solves the issue. Should I fix also this one? 
Thank you for all your suggestions :pray:

Reply:
> Why the edit mode should remove the class on save?

Just a guess but it feels more normal to save than class since this is something which depends on the visitor screensize (you cannot know when saving if the modal is always overflowing or not).

> [...] Should I fix also this one?

Well ideally, yes :) But agreed it may not be that easy... there are many ways to make the modal grow (adding snippet, resizing one, adding a lot of text, ...). Maybe listening on the content_changed event in the popup snippet option is a solution to investigate.